app/src/main/java/com/movies/common/extensions/FragmentExtension.kt

```diff
@@ -59,21 +36,41 @@ fun <T> Fragment.observeLiveData(
     }
 }
 
+fun BaseFragment<*, *>.tryConfigureBottomView(tryAction: () -> Unit) {
+    try {
+        tryAction.invoke()
+    } catch (_: Exception) {
+        (parentFragment as BaseChildFragment).addBottomContainerView(bottomView())
+    }
+}
+
 fun <T> Fragment.changeScreen(fragment: Fragment, args: T? = null) {
     args?.let {
         val bundle = Bundle().apply {
-            putArguments(args)
+            putArguments(MOVIE_ID, args)
         }
         fragment.arguments = bundle
     }
 
     parentFragmentManager.beginTransaction().add(
-        R.id.nav_host_fragment, fragment, fragment.tag
-    ).addToBackStack(fragment.tag).commit()
+        R.id.childFragmentContainerView, fragment
+    ).addToBackStack(null).commit()
 }
 
-private fun <T> Bundle.putArguments(args: T) {
+@Suppress("UNCHECKED_CAST")
+fun <T> Bundle.putArguments(key: String, args: T? = null) {
     when (args) {
-        is Int -> putInt("movieId", args)
+        is Int -> putInt(key, args)
+        is String -> putString(key, args)
+        is ArrayList<*> -> putStringArrayList(key, args as ArrayList<String>)
```

Comment:
Why do you need ArrayList? You should use kotlin's List if there is no specific exception

Reply:
done in base gradle PR

Reply:
If you want to get list or put list into the bundle you should use arrayList @DavitTsiskarashvili @bchmsl